core/domain/exp_jobs_one_off.py

```diff
@@ -56,6 +57,30 @@ def map(exploration_model):
     def reduce(exp_id, list_of_exps):
         pass
 
+class ExplorationFirstPublishedOneOffJob(jobs.BaseMapReduceJobManager):
+    """One-off job that finds first published datetime for all explorations."""
+
+    @classmethod
+    def entity_classes_to_map_over(cls):
+        return [exp_models.ExplorationRightsSnapshotContentModel]
+
+    @staticmethod
+    def map(item):
+        if item.content['status'] == rights_manager.ACTIVITY_STATUS_PUBLIC:
+            snapshot_id = item.id
+            yield(
+                snapshot_id[:snapshot_id.rfind("-")],
+                utils.get_time_in_millisecs(item.created_on))
+
+    @staticmethod
+    def reduce(exp_id, stringified_commit_times_msecs):
+        commit_times_msecs = [ast.literal_eval(commit_time_string) for
```

Comment:
Indentation:

```
commit_times_msecs = [
    ast.literal_eval(commit_time_string) for
    commit_time_string in stringified_commit_times_msecs]
```

In general, break line after ( or [ if the whole thing doesn't fit on one line.